app/controllers/concerns/cybersource.rb

```diff
@@ -0,0 +1,192 @@
+# frozen_string_literal: true
+
+# Utilities for generating and validating payment parameters for Cybersource
+module Cybersource
```

Comment:
I don't love this as a module. Most of this is business logic, not about a controller. Can we just add the two methods to the Payments controller, and move the other classes to `app/models` or `app/services` as appropriate?

Reply:
I struggled to find out how to map this to Railsisms like concern, model, etc. and opted to do this just to keep everything in one place. I'm OK moving the two methods to the controller, but what should happen to the rest? Maybe `Security` is a service, and the `PaymentRequest` and `PaymentResponse` are models?

Reply:
Yes, I like that. You could continue to keep them in the Cybersource namespace too.

Reply:
ok, extracted